xdsl/ir.py

```diff
@@ -1359,6 +1359,12 @@ def __hash__(self) -> int:
 class Region(IRNode):
     """A region contains a CFG of blocks. Regions are contained in operations."""
 
+    class Default:
+        """
+        A class to be used as a default parameter to functions when a default region
+        should be constructed.
+        """
+
```

Comment:
I'm not 100% sure this is the correct way of doing it. I'd rather see a specific sentinel value?

But then, I'm not sure how that interacts with the type system. Maybe that can be a literal of some sort?

I guess the main problem with this is that people will want to instantiate this class, so I'd recommend adding an error to the `__init__` that this class is not to be constructed. But that feels quite silly to me...

Reply:
I've seen it done this way before, and I feel like this is good enough. There's a proposal for a sentinel: https://peps.python.org/pep-0661/ it's a draft, we can use it when it gets accepted, I guess